semgrep-core/src/analyzing/Dataflow_svalue.ml

```diff
@@ -51,10 +53,77 @@ let warning tok s =
 
 let str_of_name name = spf "%s:%d" (fst name.ident) name.sid
 
+let str_of_resolved_name name =
+  let name = Common.map fst name in
+  String.concat "." name
+
 (*****************************************************************************)
 (* Constness *)
 (*****************************************************************************)
 
+let hook_constness_table_of_functions = ref None
+
+let result_of_function_call_is_constant lang f args =
+  let check_f f_name =
+    match !hook_constness_table_of_functions with
+    | Some constness_f -> (
+        match constness_f f_name with
+        | Some Constant ->
+            logger#info "%s is always constant" f_name;
+            true
+        | Some NotAlwaysConstant ->
+            logger#info "%s is not always constant" f_name;
+            false
+        | None ->
+            logger#info "%s has not been read" f_name;
```

Comment:
Nitpicking here but what does `"not been read"` mean here? I was expecting a message more like `"we have no information about the constness of %s"`.

Reply:
I was thinking of it as "we haven't read through the function body", but your wording makes sense